src/schema/v2/artwork/updateArtworkMutation.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  GraphQLString,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import Artwork from "schema/v2/artwork"
+import { omitBy } from "lodash"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+
+interface UpdateArtworkMutationInputProps {
+  id: string
+  availability?: boolean
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "updateArtworkSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    artwork: {
+      type: Artwork.type,
+      resolve: (artwork) => artwork,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "updateArtworkFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "updateArtworkResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const updateArtworkMutation = mutationWithClientMutationId<
+  UpdateArtworkMutationInputProps,
+  any,
+  ResolverContext
+>({
+  name: "UpdateArtworkMutation",
+  description: "Updates an artwork.",
+  inputFields: {
+    availability: {
+      type: GraphQLString,
+      description: "The availability of the artwork",
+    },
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The id of the artwork to update.",
+    },
+  },
+  outputFields: {
+    artworkOrError: {
+      type: ResponseOrErrorType,
+      description: "On success: the artwork note updated.",
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async ({ id, ...props }, { updateArtworkLoader }) => {
+    if (!updateArtworkLoader) {
+      return new Error("You need to be signed in to perform this action")
+    }
+
+    const payload = omitBy(props, (prop) => prop == null)
```

Comment:
This _feels_ non-standard, and if this mutation _just_ updates the artwork's availability, it might be better to just explicitly do that, vs. trying to keep this generic-y. The input arguments will necessarily need a code change if/when this mutation supports setting other properties.

Also, the removing of nulls means that, if this mutation evolves to eventually support a field that can be toggled (set and unset), it may need some special handling to 'unset' (since passing that in as a null will just get removed from the payload entirely). Overall, this is a minor line, but it feels unneeded and asking for a possible bug (at some point). The best code is no code! (ie- don't do this and _just_ handle the one `availability` field for now).